backend/uasmartsignage/src/main/java/deti/uas/uasmartsignage/Services/FileService.java

```diff
@@ -186,71 +181,196 @@ public String getParentDirectoryPath(CustomFile customFile) {
-            logger.info("Adding file to parent with path: " + customFile.getPath());
-            List<CustomFile> subDirectories = parent.getSubDirectories();
-            subDirectories.add(customFile);
-            parent.setSubDirectories(subDirectories);
-        }
+
+        // Creating file path for filesystem
+        StringBuilder pathBuilder = new StringBuilder();
+        String rootPath = System.getProperty("user.dir");
+        // Get parent path and add the filename
+        String parentDirectoryPath = getParentDirectoryPath(customFile) + fileName;
+        pathBuilder.insert(0, parentDirectoryPath);
+        pathBuilder.insert(0, rootPath);
+
+        Path fileSysPath = Paths.get(pathBuilder.toString());
+        
 
         try {
-            Files.copy(file.getFile().getInputStream(), path);
+            logger.info("Creating file with type: " + customFile.getType() + " and name: " + customFile.getName());
+            Files.copy(file.getFile().getInputStream(), fileSysPath);
+            fileRepository.save(customFile);
             return customFile;
         } 
         catch (IOException e) {
-            logger.error("Failed to copy file: {}", e.getMessage());
+            logger.error("Failed to save file: {}", e.getMessage());
             return null;
         }
     }
 
-    // TODO - need to revise logic and alike...
+    /**
+     * Deletes the desired file from the repository and the filesystem.
+     *
+     * @param id The ID of the file to delete.
+     * @return true if the file was successfully deleted, false otherwise.
+     */
+    public boolean deleteFile(Long id) {
+        Optional<CustomFile> fileOptional = fileRepository.findById(id);
+        if (fileOptional.isEmpty()) {
+            logger.warn("File with ID {} not found", id);
+            return false;
+        }
 
+        // Delete file from disk
+        CustomFile file = fileOptional.get();
+        String rootPath = System.getProperty("user.dir");
+        String filePath = rootPath + file.getPath();
+        logger.info("Deleting file: {}", filePath);
+        File fileToDelete = new File(filePath);
 
-    public CustomFile updateFile(Long id, CustomFile customFile) {
-        Optional<CustomFile> fileOptional = fileRepository.findById(id);
-            if (fileOptional.isPresent()) {
-                CustomFile file = fileOptional.get();
-                file.setName(customFile.getName());
-                file.setType(customFile.getType());
-                file.setParent(customFile.getParent());
-                file.setPath(customFile.getPath());
-                file.setSubDirectories(customFile.getSubDirectories());
-                return fileRepository.save(file);
-            } 
-            else {
-                throw new RuntimeException("File not found with id " + id);
+        if (!fileToDelete.exists()) {
+            logger.warn("File does not exist: {}", filePath);
+            return false;
+        }
```

Comment:
This check might not be needed since we already check if the file exists with its ID.